go.mod

```diff
@@ -1,27 +1,28 @@
 module github.com/weaviate/weaviate-go-client/v4
 
-go 1.21
+go 1.22
+
+toolchain go1.22.8
```

Comment:
what does this do?

Reply:
`toolchain` specifies the version of [Go Toolchain](https://www.google.com/search?client=safari&rls=en&q=go+toolchain&ie=UTF-8&oe=UTF-8) (things that you can use with `go ...` command. Usually its version is the same as the one in `go` directive, but they may differ for some versions (see **Go versions** in the link) and then we need an explicit `toolchain` directive.

I'll just set `go 1.22.0` so that the versions match and remove `toolchain`.

Reply:
From the article:

> The standard Go toolchains are named goV where V is a Go version denoting a beta release, release candidate, or release. For example, go1.21rc1 and **go1.21.0 are toolchain names**; **go1.21 and go1.22 are not** (the initial releases are go1.21.0 and go1.22.0), but **go1.20 and go1.19 are**.

It's a bit tricky: `go` and `toolchain` directives "understand" versions differently. I'm guessing that before (with go1.20 and go1.19) you could just specify the major one.
But for "toolchain" go1.22 is not a valid toolchain name.

I just stumbled upon this recently: you can try to remove `toolchain` in this go.mod and run `go version` (or any other command). You'll see this error message:

```sh
go: downloading go1.22 (darwin/amd64)
go: download go1.22 for darwin/amd64: toolchain not available
```

Reply:
I changed it to `go 1.22.0` and removed the `toolchain` directive, because that was the simplest solution I could think of  -- let me know if that works for you

Reply:
yeah works :)